validator/client/beacon-api/json_rest_handler.go

```diff
@@ -79,6 +81,13 @@ func (c *BeaconApiJsonRestHandler) Post(
 		return errors.New("data is nil")
 	}
 
+	_, hasDeadline := ctx.Deadline()
+	if !hasDeadline {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, c.timeout)
+		defer cancel()
+	}
```

Comment:
Why only set a timeout, if one isn't set already? I would expect that c.timeout sets the timeout everytime.

context.Context will cancel if the parents context has a shorter deadline anyway, so it should be safe to always set the timeout

Reply:
The idea here is to not set the timeout to `c.timeout` when the context has a deadline longer than `c.timeout`. Otherwise the request will finish too quickly. Currently we use a custom `Client` with a pre-set timeout. Even though we want to wait up to 5 minutes when pushing proposer settings, the client's timeout disallows this.

A follow-up question is what to do when a context's deadline is less than `c.timeout`. This code will respect the context's timeout.